index.js

```diff
@@ -0,0 +1,259 @@
+const navWrapper = document.querySelector(".nav-wrapper");
+const navToggle = document.querySelector(".mobile-nav-toggle");
+const primaryNavigation = document.querySelector(".primary-navigation");
+
+document.body.addEventListener("click", (e) => {
+  if (
+    !navWrapper.contains(e.target) &&
+    navToggle.attributes[2].nodeValue === "true"
+  ) {
+    closeNav();
+  }
+});
+navWrapper.addEventListener("click", (e) => {
+  if (navToggle.attributes[2].nodeValue === "false") {
+    showNav();
+  } else {
+    closeNav();
+  }
+});
+
+const showNav = () => {
+  navToggle.setAttribute("aria-expanded", true);
+  primaryNavigation.toggleAttribute("opening");
+  primaryNavigation.addEventListener(
+    "animationend",
+    () => {
+      primaryNavigation.toggleAttribute("opened");
+      primaryNavigation.toggleAttribute("opening");
+    },
+    { once: true }
+  );
+};
+
+const closeNav = () => {
+  primaryNavigation.toggleAttribute("closing");
+  navToggle.setAttribute("aria-expanded", false);
+  primaryNavigation.addEventListener(
+    "animationend",
+    () => {
+      primaryNavigation.toggleAttribute("opened");
+      primaryNavigation.toggleAttribute("closing");
+    },
+    { once: true }
+  );
+};
+
+const contactMeButton = document.querySelector("#contact-me");
+const modal = document.querySelector("#modal");
+const closeModalButton = document.querySelector(".modal__close");
+const mobileContactButton = document.querySelector(".button--mobile");
```

Comment:
Nit: `querySelector` always scares me, since it gives you the first matching element. There could be multiple `modal__close` classes. It's just a concern if you're gonna have multiple modals. This is fine, no worries just $.03